### ISSUE TITLE
Unhandled memory fault when executing plot1 two times in a row

### ISSUE THREAD

**Issue by asikstel** [mmontone/burgled-batteries.syntax#3]
Hi,

continuing improving this great dist.

Like the title says:

# startup python, etc

BURGLED-BATTERIES> (plot1)
# works fine
BURGLED-BATTERIES> (plot1)
DEBUGGER:
Unhandled memory fault at #x1.
   [Condition of type SB-SYS:MEMORY-FAULT-ERROR]

Restarts:
 0: [RETRY] Retry SLIME REPL evaluation request.
 1: [*ABORT] Return to SLIME's top level.
 2: [ABORT] abort thread (#<THREAD "repl-thread" RUNNING {10053B0033}>)

Backtrace:
  0: (SB-SYS:MEMORY-FAULT-ERROR)
  1: ("foreign function: call_into_lisp")
  2: ("foreign function: post_signal_tramp")
  3: ("foreign function: #x2011718BAA")
  4: ((SB-PCL::EMF RUN*) #<unavailable argument> #<unavailable argument> "import matplotlib.pyplot as plt")
  5: (RUN "import matplotlib.pyplot as plt")
  6: ((FLET #:CLEANUP-FUN-0 :IN PLOT1)) [cleanup]
  7: (PLOT1)
  8: (SB-INT:SIMPLE-EVAL-IN-LEXENV (PLOT1) #<NULL-LEXENV>)
  9: (EVAL (PLOT1))
 --more--

That thing doesnt happen if I work without the syntax, i.e. only in burgled-batteries package. 
I think that importing the library is the problem. Somehow it seems that the function plot1 doesnt clean the symbol in the namespace which holds the matplotlib library after finishing. That also happens with other libraries you try to import inside a function.
I couldn't locate that error more precisely unfortunately. 

Best regards,
Aleksey

**Comment by mmontone** [comment]
Yep. I'm aware of the bug, but I don't know how to fix it. Maybe it has to do with symbols namespaces, like you say. I don't know

**Comment by asikstel** [comment]
Hi,

I've found out that if you don't try to kill python inside the defun form there are no errors.
Hence as a crutch I'll define a macro which starts burgled batteries etc and kills it afterwards. This I'll wrap around all plots I need. But it's only a crutch. And it seems to be more an error of burgled-batteries rather than the syntax part.
Best

**Comment by mmontone** [comment]
Ok. So it seems you are not supposed to start and stop Python several times. You have to start it once and work with it as long as you can. Sounds bearable, although I would like to know the reason it does not work.
